prism/src/main/java/org/apache/falcon/service/EntitySLAAlertService.java

```diff
@@ -142,12 +142,18 @@ void processSLACandidates(){
                     LOG.info("Entity : {} Cluster : {} Nominal Time : {} missed SLALow", entityName, entityType,
                             clusterName, nominalTime);
                 } else if (schedulableEntityInstance.getTags().contains(EntitySLAMonitoringService.get().TAG_CRITICAL)){
-                    if (entityType.equalsIgnoreCase(EntityType.PROCESS.name())){
-                        store.putSLAAlertInstance(entityName, clusterName, entityType,
-                                nominalTime, true, false);
+                    if (entityType.equalsIgnoreCase(EntityType.PROCESS.name())) {
+                        if (store.getAlertInstancesList(entityName, clusterName, nominalTime, entityType).size() == 0){
```

Comment:
Basically, you need a putIfAbsent kind of functionality. It would be better to add such a method to the store itself.